console/src/view/table.rs

```diff
@@ -0,0 +1,183 @@
+use crate::{
+    input, state,
+    view::{self, bold},
+};
+use std::convert::TryFrom;
+use tui::{
+    layout,
+    text::{self, Span, Spans, Text},
+    widgets::TableState,
+};
+
+use std::cell::RefCell;
+use std::rc::Weak;
+
+pub(crate) trait TableList<R, S: TryFrom<usize>> {
+    fn header(&self) -> &'static [&'static str];
```

Comment:
it seems like this could probably be an associated constant, rather than a method, since the headers are not going to change dynamically:
```suggestion
    const HEADER: &'static [&'static str];
```